controllers/user/password.js

```diff
@@ -0,0 +1,240 @@
+const xss = require('xss');
+const jwt = require('jsonwebtoken');
+const { validationResult } = require('express-validator/check');
+
+const models = require('../../models/index');
+const mail = require('../../modules/mail');
+const secrets = require('../../modules/secrets');
+const { userBasePath, userEndpoints, makeUserLink, renderUserError } = require('./utils');
+
+const requestResetForm = (req, res) => {
+    res.cacheControl = { maxAge: 0 };
+    res.render('user/resetpassword', {
+        mode: 'enterEmail',
+        makeUserLink: makeUserLink
+    });
+};
+
+// is this user in password update mode?
+const checkUserRequestedPasswordReset = (userId, callbackSuccess, callbackError) => {
+    models.Users
+        .findOne({
+            where: {
+                id: userId,
+                is_password_reset: true
+            }
+        })
+        .then(user => {
+            if (!user) {
+                // no user for this ID, or they already did this reset
+                return callbackError();
+            } else {
+                return callbackSuccess(user);
+            }
+        })
+        .catch(err => {
+            callbackError(err);
+        });
+};
+
+const changePasswordForm = (req, res) => {
+    res.cacheControl = { maxAge: 0 };
+    let token = req.query.token;
+    if (!token) {
+        return res.redirect(userBasePath + userEndpoints.login);
+    }
+
+    // a user has followed a reset link
+    jwt.verify(token, secrets['user.jwt.secret'], (err, decoded) => {
+        if (err) {
+            console.error('Password reset token expired', err);
```

Comment:
Do these console logs still need to be in here? If you are interesting in tracking this we could send an info level event to Sentry instead?

Reply:
Hmm yeah, good point – will swap these out.